clients/src/main/java/org/apache/kafka/common/record/FileLogInputStream.java

```diff
@@ -98,6 +100,7 @@ public FileChannelRecordBatch nextBatch() throws IOException {
     public abstract static class FileChannelRecordBatch extends AbstractRecordBatch {
         protected final long offset;
         protected final byte magic;
+        protected final File file;
```

Comment:
If we need both the `File` and `FileChannel`, I wonder if we may as well pass a reference to `FileRecords`. We can get to the channel using `FileRecords.channel()` and we can rely on `FileRecords.toString()` for a nicer representation in logs and exception messages.

Reply:
Good point 🙂
I published an update to address this. I had to add implementations for `FileRecords.{equals, hashCode}` (they're just leveraging the internal `channel`, to mimic the previous behaviour). Let me know if you see any issue with this.